Lib/fontTools/feaLib/builder.py

```diff
@@ -500,156 +499,160 @@ def checkattr(table, attrs):
+                )
         self.stat_["ElidedFallbackName"] = names
 
-    def set_ElidedFallbackNameID(self, value):
+    def setElidedFallbackNameID(self, value, location):
+        for token in ("ElidedFallbackName", "ElidedFallbackNameID"):
+            if token in self.stat_:
+                raise FeatureLibError(
+                    f"{token} is already set.",
+                    location,
+                )
         self.stat_["ElidedFallbackNameID"] = value
 
-    def addDesignAxis(self, location, key, axisOrder, value):
+    def addDesignAxis(self, designAxis, location):
         if "DesignAxes" not in self.stat_:
-            self.stat_["DesignAxes"] = {}
-        if key in self.stat_["DesignAxes"]:
+            self.stat_["DesignAxes"] = []
+        if designAxis.tag in (r.tag for r in self.stat_["DesignAxes"]):
+            raise FeatureLibError(
+                'DesignAxis already defined for tag "%s".' % designAxis.tag,
+                location,
+            )
+        if designAxis.axisOrder in (r.axisOrder for r in self.stat_["DesignAxes"]):
             raise FeatureLibError(
-                'DesignAxis already defined for tag "%s".' % key,
+                f"DesignAxis already defined for axis number {designAxis.axisOrder}.",
                 location,
             )
-        self.stat_["DesignAxes"][key] = (axisOrder, value)
+        self.stat_["DesignAxes"].append(designAxis)
 
-    def addAxisValueRecord(self, location, names, locations, flags):
+    def addAxisValueRecord(self, axisValueRecord, location):
         if "AxisValueRecords" not in self.stat_:
             self.stat_["AxisValueRecords"] = []
         # Check for duplicate AxisValueRecords
-        for record in self.stat_["AxisValueRecords"]:
-            names_, locations_, flags_ = record
-            if sorted(names_) == sorted(names) and sorted(locations_) == \
-                    sorted(locations) and sorted(flags_) == sorted(flags):
+        for record_ in self.stat_["AxisValueRecords"]:
+            if (sorted([n.asFea() for n in record_.names]) ==
+                sorted([n.asFea() for n in axisValueRecord.names]) and
+                sorted(record_.locations) == sorted(axisValueRecord.locations)
+                    and record_.flags == axisValueRecord.flags):
                 raise FeatureLibError(
-                    'Duplicate AxisValueRecords are not allowed.',
+                    "An AxisValueRecord with these values is already defined.",
                     location,
                 )
-        self.stat_["AxisValueRecords"].append((names, locations, flags))
+        self.stat_["AxisValueRecords"].append(axisValueRecord)
 
     def build_STAT(self):
         if not self.stat_:
             return
         self.font["STAT"] = newTable("STAT")
         table = self.font["STAT"].table = otTables.STAT()
         table.Version = 0x00010001
-        name_table = self.font.get("name")
+        nameTable = self.font["name"]
```

Comment:
in bare-bone fonts (e.g. unit tests) the name table may not be present, here's what we do elsewhere in feaLib.builder

https://github.com/fonttools/fonttools/blob/a41005077dc36f27798e64d659934aa3dcdf016a/Lib/fontTools/feaLib/builder.py#L391-L394

Reply:
Replaced